app/models/rides.py

```diff
@@ -1,62 +1,222 @@
-		return True
-	
-	def fetch_all(self,items):
-		if len(items) > 0:
-			return True
-		else:
-			return False
-	
-	def fetch_one(self,id,items):
-		index = 0
-		while (index < len(items)):
-			if int(id) == items[index]['id']:
-				return True
-			index = index + 1
-		
-
-	def request_ride(self,id,name,time,destination,pickup,items):
-		index=0
-		data ={
-		'name':name,
-		'time':time,
-		'destination':destination,
-		'pickup':pickup,
-		'id':len(self.all_request)+1
+		rides.ride_data.append(item)
+		return {'result':'Content Uploaded'},201
+
+	def ride_exist(self,id):
+		id = int(id)
+		if id > len(rides.ride_data):
+			return False
+		elif id < 0:
+			return False
+		else:
+			return True
+
+	def view_ride_single(self,id):
+		return rides.ride_data[int(id)]
+
+	def key_format(self,items):
+		items = items.keys()
+		if 'username' not in items:
+			return False
+		elif 'capacity' not in items:
+			return False
+		elif 'time' not in items:
+			return False
+		elif 'date' not in items:
+			return False
+		elif 'month' not in items:
+			return False
+		elif 'year' not in items:
+			return False
+		elif 'pickup' not in items:
+			return False
+		elif 'destination' not in items:
+			return False
+		elif 'email' not in items:
+			return False
+		else:
+			return True
+
+	def is_string(self,input):
+		if isinstance(str(input['username']),str):
+			return True
+		elif isinstance(input['time'],str):
+			return True
+		elif isinstance(input['date'],str):
+			return True
+		elif isinstance(input['month'],str):
+			return True
+		elif isinstance(input['destination'],str):
+			return True
+		elif isinstance(input['email'],str):
+			return True
+		else:
+			return False
+
+	def is_integer(self,input):
+		if isinstance(int(input['year']),int):
+			return True
+		elif isinstance(int(input['capacity']),int):
+			return True
+		else:
+			return False
+
+	def same_direction(self,pickup,destination):
+		if pickup == destination:
+			return True
+		else:
+			return False
+	def is_empty(self,input):
+		if len(input['username']) == 0:
+			return True
+		elif len(input['time']) == 0:
+			return True
+		elif len(input['date']) == 0:
+			return True
+		elif len(input['month']) == 0:
+			return True
+		elif len(input['pickup']) == 0:
+			return True
+		elif len(input['destination']) == 0:
+			return True
+		elif len(input['email']) == 0:
+			return True
+		else:
+			return False
+
+	def only_space(self,input):
+		if input['username'].isspace():
+			return True
+		elif input['time'].isspace():
+			return True
+		elif input['month'].isspace():
+			return True
+		elif  input['pickup'].isspace():
+			return True
+		elif input['destination'].isspace():
+			return True
+		elif input['email'].isspace():
+			return True
+		elif input['date'].isspace():
+			return True
+		else:
+			return False
+
+	def correct_data(self,input):
+		months = ['january', 'jebruary', 'march', 'april', 'may', 'june', 'july', 'august',
+		 'september', 'october', 'november', 'december']
```

Comment:
- On dates, consider [Python Datetime](https://docs.python.org/3/library/datetime.html) module.
- I don't think it's a good idea to hardcode this.
- For example, you have an `elif` referring to 2018. Don't you think this project should be valid in 2019?